share/availability/discovery/set.go

```diff
@@ -14,12 +13,12 @@ type limitedSet struct {
 	lk sync.RWMutex
 	ps map[peer.ID]struct{}
 
-	limit    uint
+	limit    int
 	waitPeer chan peer.ID
 }
 
 // newLimitedSet constructs a set with the maximum peers amount.
-func newLimitedSet(limit uint) *limitedSet {
+func newLimitedSet(limit int) *limitedSet {
 	ps := new(limitedSet)
```

Comment:
if you've changed from uint to int, then I'd propose adding a sanity check for` limit > 0`

Reply:
We allow -1 as a way to disable discovery

Reply:
May be we should also add comment here, that explains 0 limit case?

Reply:
Making it uint again

Reply:
ahhh i dislike but its ok.